src/modules/course/course.service.ts

```diff
@@ -156,4 +164,74 @@ export class CourseService {
       throw new InternalServerErrorException(e);
     }
   }
+
+  // Create a Review
+  async addReview(
+    courseId: Schema.Types.ObjectId,
+    createReviewDto: CreateReviewDto,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        const newReview = new this.ReviewModel(createReviewDto);
+        await newReview.save();
+        course.reviews.push(newReview);
+        await course.save();
+        return newReview;
+      } else {
+        throw new NotFoundException(
+          'The course id is invalid or the course no longer exists',
+        );
+      }
+    } catch (e) {
+      throw new InternalServerErrorException(e);
+    }
+  }
+
+  // update a Review by Id
+  async updateReview(
+    courseId: Schema.Types.ObjectId,
+    reviewId: Schema.Types.ObjectId,
+    updateReviewDto: UpdateReviewDto,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        let updatedReview = null;
+        updatedReview = await this.ReviewModel.findByIdAndUpdate(
+          reviewId,
+          updateReviewDto,
+          { new: true },
+        );
+        return updatedReview;
+      } else {
+        throw new NotFoundException(
+          'The course id is invalid or the course no longer exists',
+        );
+      }
+    } catch (e) {
+      throw new InternalServerErrorException(e);
+    }
+  }
+
+  // Delete a Review by Id
+  async deleteReview(
+    courseId: Schema.Types.ObjectId,
+    reviewId: Schema.Types.ObjectId,
+  ): Promise<any> {
+    try {
+      const course = await this.CourseModel.findById(courseId);
+      if (course) {
+        let deletedReview = null;
+        deletedReview = await this.ReviewModel.findByIdAndRemove(reviewId);
+        return deletedReview;
```

Comment:
Do the same for this and test it in postman

Reply:
Done, It returns the same correct object